pkg/providers/imagefamily/nodeimage.go

```diff
@@ -44,7 +44,7 @@ func (p *Provider) List(ctx context.Context, nodeClass *v1beta1.AKSNodeClass) ([
 		return []NodeImage{}, err
 	}
 
-	supportedImages := getSupportedImages(nodeClass.Spec.ImageFamily)
+	supportedImages := getSupportedImages(nodeClass.Spec.ImageFamily, kubernetesVersion)
```

Comment:
Note: this will trigger any customer currently using AzureLinux and on k8s version 1.32.x or later to Drift and roll all of their nodes when we release this.

Reply:
https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar

1.32 is already GA:
![image](https://github.com/user-attachments/assets/75b5ef93-6cfb-4a57-891d-93cde8abdc98)

Reply:
Ack. Still the right thing to do IMO, to align with AKS.